spec/support/prefabricated.rb

```diff
@@ -0,0 +1,44 @@
+module Prefabricated
+  def setup_prefabrication
+    @prefabricated_things = {}
+    @prefabricated_descriptors = []
+    @prefabricated_classes = {}
+    @prefabricated_ids = {}
+
+    prefabricated_things = @prefabricated_things
+    prefabricated_descriptors = @prefabricated_descriptors
+    prefabricated_classes = @prefabricated_classes
+    prefabricated_ids = @prefabricated_ids
+
+    before(:each) do
+      prefabricated_things.clear
+    end
+
+    before(:all) do
+      ActiveRecord::Base.connection.begin_transaction(joinable: false)
+      prefabricated_descriptors.each do |name, generator|
+        thing = generator.call
+        prefabricated_classes[name] = thing.class
+        prefabricated_ids[name] = thing.id
+        prefabricated_things[name] = thing
+      end
+    end
+
+    after(:all) do
+      ActiveRecord::Base.connection.rollback_transaction
+    end
+  end
+
+  def prefabricate(name, &blk)
+    setup_prefabrication unless @prefabricated_things
+
+    @prefabricated_descriptors << [name, blk]
+    prefabricated_things = @prefabricated_things
+    prefabricated_classes = @prefabricated_classes
+    prefabricated_ids = @prefabricated_ids
+
+    define_method(name) do
```

Comment:
Wouldn't this leak the methods into other examples?

Reply:
I don't believe so. RSpec creates a class for each context and define_method puts the method on the class, not the module, as illustrated here:

```ruby
module Test
  def foo
    define_method(:bar) do
      :bar
    end
  end
end

class Foo
  extend Test

  foo
end

class Bar
  extend Test
end

p Foo.new.respond_to?(:bar) # => true
p Bar.new.respond_to?(:bar) # => false
```

Reply:
I've added specs for fab! to demonstrate that it does what you'd expect w.r.t scoping.